### ISSUE TITLE
If field asks for input about "your organization" is that covered under SC 1.3.5?

### ISSUE THREAD

**Issue by mraccess77** [w3c/wcag#642]
If a field asks for "your organization's zipcode" does that fall outside of SC 1.3.5 Input purpose?  Seems like an organization would -- but since it says "your organization" it not as clear and seems like something we should document the agreed upon answer.

**Comment by alastc** [comment]
From a quick discussion on the call, my suggestion (not approved) was:

No, there is a value that we equate to "your organisation", but the address fields would be required for you (unspecified).

However, if people do use autocomplete attributes for address on organisational inputs, that is not a failure.

I.e. We should allow for multiple profiles in people's browsers.

**Comment by awkawk** [comment]
Proposed WG Response: 1.3.5 Identify Purpose is designed to apply to input fields about the user and this includes some organizational information (specifically Title and Organization name) that applies directly to the user. Additional information about the organization such as the Address or Postal code for the organization does not apply specifically to the user and is not required for conformance. 

Making use of additional aspects for a technology, where specified, to include autocomplete attribute values beyond items that apply directly to the user is encouraged, but not required by SC 1.3.5 Identify Purpose.

**Comment by mraccess77** [comment]
Thanks, to clarify a related point -- organization address and zip would not be -- but work address or work phone would be -- is that correct?

**Comment by awkawk** [comment]
I don't think that we require that the form differentiate between home and work addresses for an individual. Others?

**Comment by alastc** [comment]
The meta-data values don't differentiate, so our only additional requirement is that it is "about the user".

My read is that: If it is an address field about the user, I.e. "your address" as an individual it doesn't matter if it is work, home, holiday apartment or postal box; it is required.

Work address and organizational address may not be the same thing, you might be in a different building to the official org address. Therefore if it is asking for a organizational (or even "your organization's") address then it is not required.

I think we have to leave it to the author's intent rather than what a user would perceive, that is clearer and more knowable.

**Comment by mraccess77** [comment]
@alastc I agree.  Thank you.  I accept this thread/response.

**Comment by awkawk** [comment]
@alastc ok, so in HTML for an input, what would the autocomplete attribute value need to be for "Work postal code"?

**Comment by mraccess77** [comment]
I thought @alastc was just saying "postal-code" would be required for work postal code and for home postal code if they are both requesting information about the user.

**Comment by awkawk** [comment]
@mraccess77 that's what I think I was saying - I want to verify if that is what @alastc is saying also! :)

**Comment by alastc** [comment]
> so in HTML for an input, what would the autocomplete attribute value need to be for "Work postal code"?

If it is asking for work postal code, that is not (directly) about the user, so the autocomplete attr it is not required, but could be used.

If it is asking for *your* postal code, which may or may not be a work one, then it is required.

**Comment by awkawk** [comment]
@alastc you said above: "My read is that: If it is an address field about the user, I.e. "your address" as an individual it doesn't matter if it is work, home, holiday apartment or postal box; it is required."

I'm assuming that your work address also has a postal code, how would you mark up the autocomplete attribute? AND, what do you put if the form asks for both home and work addresses?

**Comment by alastc** [comment]
`<input id="postalcode" type="text" autocomplete="postal-code">`

This is coming down to whether you consider the work address as about the user or not, and I could go either way on "your work address". If the intent is to post something to your organization, then that is the target. If the intent is to post something to you, at that address, then it's about you.

My main point is that where it's about the user it is required, where it is not directly about the user it is allowed. 

Where there are two addresses for _you_, I'd probably put autocomplete on both. As there's no differentiation in the metadata, authors can't specify which is which.

**Comment by johnfoliot** [comment]
Alastair wrote:

> 
> <input id="postalcode" type="text" autocomplete="postal-code">
> 
> Where there are two addresses for *you*, I'd probably put autocomplete on
> both. As there's no differentiation in the metadata, authors can't specify
> which is which.

As a practical matter, if the intent of the form is to capture  the
postal-code of you, at HOME as well as at WORK, and those values differ,
then the autocomplete values (used for that function) stored on the browser
will only consist of *one* value (postal-code=12345), with the *presumption*
that it would be the home address. Placing @autocomplete on *both *of those
fields would likely result in the answer being wrong 50% of the time when
values are stored on a browser (which is not always a given for every
user-agent). A certain level of logic must prevail here due to the current
technique and how the auto-completing is supported by current browsers, and
so I would NOT place @autocomplete on both inputs, only on the home
postal-code input. (It's *MY* postal code, not my *WORK*'s Postal Code; the
work postal-code is one "layer" removed from the user).

**Comment by awkawk** [comment]
Of course, there would still be a benefit if a user agent augmented the field label with a graphic that helped the user understand that the field is asking for a postal code, but there would also be potential confusion if a user agent always replaced the full label for that control with a specific text string and replaced "work zip code" and "home zip code" labels with "Postal Code" (just an example).

**Comment by mraccess77** [comment]
@johnfoliot If a site asks about my work address -- say for emergency planning purposes -- that seems very much information about me, the user.    I could live with it either way -- but we need to build a consensus on what people accept is a failure or not as the difference can be a "partially supports" vs. a "supports" on an accessibility conformance report.

**Comment by alastc** [comment]
> It's *MY* postal code, not my *WORK*'s Postal Code; the work postal-code is one "layer" removed from the user

It's tricky, how about people in care homes, institutions, living with family... they might not have any other address without that layer, it's a hard line to draw.

Perhaps the simplest option is that anything aimed at the user, with implication for where you personally live (or phone number etc.) ALSO, using it autocomplete on other fields would not be a fail.

**Comment by patrickhlauke** [comment]
noting that, speaking strictly about HTML here, it is possible to scope the same autocomplete value into different groups

> Some autocomplete values may need to be re-used multiple times. For example, a form may contain multiple shipping addresses and therefore multiple occurrences of "shipping postal-code" while still expecting different values. To make the autocomplete value unique in these cases, the first token in the space-separated list of tokens can be a section-* token, where the token's first eight characters are always the string "section-", followed by an alphanumeric string. All form fields given the section-* token with the same alphanumeric string belong to the same named group.

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/autocomplete

however, this is only for HTML, and may not be a universal concept. normatively, WCAG only requires that form controls that ask for information about the user be programmatically identified. it doesn't make any further normative distinction here, and it's left up to the author / auditor to judge whether a field is really about the user, or some other aspect ... which i don't think can be resolved further unless the normative wording becomes a lot stricter/covers more aspects that go beyond what one might expect to be about the user in a strict sense.

**Comment by patrickhlauke** [comment]
**Working group response:** the scoping of 1.3.5 Identify Input Purpose is slightly handwavy/vague when referring to "information about the user". It may not be possible to be more explicit/specific and cover aspects such as the one mentioned in this thread (is a user's organisation's information also "about the user"?). Generally, this has been interpreted as "personal information".
